python/uyuni-cobbler-helper/uyuni-cobbler-helper.spec

```diff
@@ -0,0 +1,52 @@
+#
+# spec file for package uyuni-cobbler-helper
+#
+# Copyright (c) 2024 SUSE LLC
+#
+# All modifications and additions to the file contributed by third parties
+# remain the property of their copyright owners, unless otherwise agreed
+# upon. The license for this file, and modifications and additions to the
+# file, is the same license as for the pristine package itself (unless the
+# license for the pristine package is not an Open Source License, in which
+# case the license is the MIT License). An "Open Source License" is a
+# license that conforms to the Open Source Definition (Version 1.9)
+# published by the Open Source Initiative.
+
+# Please submit bugfixes or comments via https://bugs.opensuse.org/
+#
+
+%{!?python3_sitelib: %global python3_sitelib %(%{__python3} -c "from distutils.sysconfig import get_python_lib; print(get_python_lib())")}
+
+
+Name:           uyuni-cobbler-helper
+Version:        0.1.0
+Release:        0
+Summary:        Python helper functions for Uyuni cobbler snippets
+License:        Apache-2.0
+Group:          System/Management
+URL:            https://github.com/uyuni-project/uyuni
+Source0:        %{name}-%{version}.tar.gz
+BuildRoot:      %{_tmppath}/%{name}-%{version}-build
+%if !0%{?suse_version} || 0%{?suse_version} >= 1120
+BuildArch:      noarch
+%endif
+
+Requires:       python3
+Requires:       python3-psycopg2 >= 2.8.4
```

Comment:
Be aware that the image seems to be installing both `python3-psycopg2` and `python311-psycopg2` in parallel, is it desired?
```
[   37s] Retrieving: python3-psycopg2-2.8.5-5.7.1.x86_64 (obs_repository) (280/673), 174.8 KiB    
[...]
[   37s] Retrieving: python311-psycopg2-2.9.7-150400.12.3.1.x86_64 (obs_repository) (327/673), 202.6 KiB    
```

Reply:
Should I use this?
```
%if 0%{?suse_version} == 1500 && 0%{?sle_version} >= 150700
%global use_python python311
%else
%global use_python python3
%endif
```

Reply:
I am not 100% sure how to handle this for python packages, but forcing the version like that doesn't seem right, as a future SLE version can have python312, and we'd need to keep extending this.

Maybe you can check with Ion, how they fixed it, as I think they have packages using the python latest version available on the OS, somehow.

Or maybe I am wrong and this is how it's done :)

Reply:
for now this is what we have on salt and other packages where we need to keep aligned  the python version.

Reply:
I opened https://github.com/cbosdo/uyuni/pull/3

From what I can tell, this helper is imported by Cobbler. That means we need it for the Python flavor Cobbler uses, because each Python flavor has a separate import path. I would just build this helper for all flavors, and when we install it we can either pick a particular flavor, or I can add a flavor-agnostic symbol we can install and let zypper resolve it to the right version that fits the dependencies

Reply:
I just updated the PR with a `Provides: uyuni-cobbler-helper`, this works now: `zypper install uyuni-cobbler-helper`
```

~uyuni/uy-work[uyuni-cobbler-helper-packaging]
% zin uyuni-cobbler-helper
[..]
'uyuni-cobbler-helper' not found in package names. Trying capabilities.
Resolving package dependencies...

The following 2 NEW packages are going to be installed:
  python311-psycopg2 python311-uyuni-cobbler-helper

2 new packages to install.
[...]
```